improver/wxcode/modal_code.py

```diff
@@ -191,6 +196,8 @@ def process(self, cubes: CubeList) -> Cube:
         """
         # Handle case in which a single time is provided.
         if len(cubes) == 1:
```

Comment:
I think that the special case is getting too special. It also leaves open the possibility that the time point on the input cube is not at the upper bound, which `_set_blended_times` would change. I've tested this alternative code (for the whole process method) and the unit and acceptance tests still pass.

```
        # Handle case in which a single time is provided.
        cube = MergeCubes()(cubes)
        self._unify_day_and_night(cube)

        if len(cube.coord("time").points) == 1:
            result = cube
            result.add_cell_method(self.mode_cell_method)
        else:
            result = cube.collapsed("time", self.aggregator_instance)
        self._set_blended_times(result)

        # Handle any unset points where it was hard to determine a suitable mode
        if (result.data == UNSET_CODE_INDICATOR).any():
            self._group_codes(result, cube)

        return result
```

Reply:
Fine by me, it's just rearrangement. The time stuff should never be an issue unless something has gone wrong with the inputs, but this ensures against it I guess.